scripts/Metacalibration/metacal.py

```diff
@@ -0,0 +1,189 @@
+"""
+Implements the basic process of metacalibration.
+Calculates the R shear response matrix from a galaxy
+"""
+
+import galsim
+import numpy as np
+
+
+ 
+def generate_observed_galaxy(source_galaxy, psf_blur, lensing_g1, lensing_g2):
+	"""
+	Takes in a source_galaxy and psf_blur GalSim objects as well as the parameters
+	g1 and g2 of the lensing shear and generates a new GalSim object representing
+	an observed galaxy
+	"""
+	# shearing the original galaxy
+	sheared = source_galaxy.shear(g1=lensing_g1, g2=lensing_g2)
+
+	# Convolving the sheared galaxy with the PSF
+	observed = galsim.Convolve(sheared, psf_blur)
+
+	return observed
+
+
+def delta_shear(observed_gal, psf_deconvolve, psf_reconvolve, delta_g1, delta_g2):
+	"""
+	Takes in an observed galaxy object, two PSFs for metacal (deconvolving
+	and re-convolving), and the amount by which to shift g1 and g2, and returns
+	a tuple of tuples of modified galaxy objects.
+	((g1plus, g1minus), (g2plus, g2minus))
+	"""
+	# Deconvolving by psf_deconvolve
+	inv_psf = galsim.Deconvolve(psf_deconvolve)
+	deconvolved = galsim.Convolve(observed_gal, inv_psf)
+
+	# Applying second shear in g1
+	sheared_plus_g1 = deconvolved.shear(g1=delta_g1, g2=0)
+	sheared_minus_g1 = deconvolved.shear(g1=-delta_g1, g2=0)
+
+	# Applying second shear in g2
+	sheared_plus_g2 = deconvolved.shear(g1=0, g2=delta_g2)
+	sheared_minus_g2 = deconvolved.shear(g1=0, g2=-delta_g2)
+
+	# Reconvolving by psf_reconvolve for g1
+	reconvolved_plus_g1 = galsim.Convolve(sheared_plus_g1, psf_reconvolve)
+	reconvolved_minus_g1 = galsim.Convolve(sheared_minus_g1, psf_reconvolve)
+
+	g1_plus_minus = (reconvolved_plus_g1, reconvolved_minus_g1)
+
+
+	# Reconvolving by psf_reconvolve for g2
+	reconvolved_plus_g2 = galsim.Convolve(sheared_plus_g2, psf_reconvolve)
+	reconvolved_minus_g2 = galsim.Convolve(sheared_minus_g2, psf_reconvolve)
+
+	g2_plus_minus = (reconvolved_plus_g2, reconvolved_minus_g2)
+
+	# g1_plus_minus = (sheared_plus_g1, sheared_minus_g1)
+	# g2_plus_minus = (sheared_plus_g2, sheared_minus_g2)
+
+	# adding noshear reconvolved for testing
+	reconvolved_noshear = galsim.Convolve(deconvolved, psf_reconvolve)
+
+	return g1_plus_minus, g2_plus_minus, reconvolved_noshear
+
+
+def shear_response(g1_plus_minus, g2_plus_minus, reconvolved_noshear, cshear_delta_g1, cshear_delta_g2, psf_shearestimator, shearestimator, pixel_scale): #TODO change this back to 0.2
+	"""
+	Takes in the a tuple of the g1 plus/minus objects and
+	a tuple of the g2 plus/minus objects and returns the
+	shear response matrix R
+	"""
+
+	plus_g1_gal = g1_plus_minus[0]
+	minus_g1_gal = g1_plus_minus[1]
+	plus_g2_gal = g2_plus_minus[0]
+	minus_g2_gal = g2_plus_minus[1]
+
+	# Measuring galaxy shape parameters
+	# We want to measure the shapes of reconvolved_plus_galaxy and reconvolved_minus_galaxy
+	# the documentation recommends that we use the method='no_pixel' on the images
+
+	plus_g1 = plus_g1_gal.drawImage(scale=pixel_scale, method='no_pixel')
```

Comment:
Why do you think this is better to draw with the 'no_pixel' method? By letting the 'no_pixel' method, you let GalSim choose the size of the image and it will depend on the operation that you applied to the image. I would rather do something like that:

```suggestion
	plus_g1 = plus_g1_gal.drawImage(nx=40, ny=40, scale=pixel_scale)
```

Reply:
I would apply this everywhere the `drawImage` is used, like that this is everywhere consistent